src/proxy/outbound.rs

```diff
@@ -410,6 +416,14 @@ impl OutboundConnection {
         if us.workload.gateway_address.is_none() {
             return Err(Error::NoGatewayAddress(Box::new(us.workload.clone())));
         }
+
+        let workload_ip = self
+            .pi
+            .state
+            .state
+            .read()
+            .unwrap()
+            .load_balance(&us.workload)?;
```

Comment:
This usage is not great - may be nicer to have load_balance on DemandProxyState?

I think we could probably get to a point where `state` is not `pub` in `DemandProxyState`

Reply:
yeah I think it will have to move, since `load_balance()` will need to be async for on-demand DNS queries. and the rust async stuff wasn't loving async functions on the proxy state (understandably.. we don't want to lock our store and then await a DNS query so no one else can read in the meantime).

anyways this will require a bit of refactoring to get to compile and be reasonable, and that's my current focus